apps/frontend/src/components/RampSubmitButton/RampSubmitButton.tsx

```diff
@@ -27,16 +29,27 @@ const useButtonContent = ({ toToken, submitButtonDisabled }: UseButtonContentPro
   const { t } = useTranslation();
   const rampActor = useRampActor();
   const stellarData = useStellarKycSelector();
+  const { address: accountAddress } = useVortexAccount();
 
-  const { isQuoteExpired, rampState, rampPaymentConfirmed, machineState, quote } = useSelector(rampActor, state => ({
-    isQuoteExpired: state.context.isQuoteExpired,
-    machineState: state.value,
-    quote: state.context.quote,
-    rampPaymentConfirmed: state.context.rampPaymentConfirmed,
-    rampState: state.context.rampState
-  }));
+  const { isQuoteExpired, rampState, rampPaymentConfirmed, machineState, walletLocked, quote } = useSelector(
+    rampActor,
+    state => ({
+      isQuoteExpired: state.context.isQuoteExpired,
+      machineState: state.value,
+      quote: state.context.quote,
+      rampPaymentConfirmed: state.context.rampPaymentConfirmed,
+      rampState: state.context.rampState,
+      walletLocked: state.context.walletLocked
+    })
+  );
 
   return useMemo(() => {
+    if (walletLocked && accountAddress && getAddressForFormat(accountAddress, 0) !== walletLocked) {
```

Comment:
We would need to normalize both addresses, just to be safe. 
```suggestion
    if (walletLocked && accountAddress && getAddressForFormat(accountAddress, 0) !== getAddressForFormat(walletLocked, 0)) {
```